omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/WristbandFragmentController.java

```diff
@@ -33,19 +37,26 @@ public SimpleObject printWristband(UiUtils ui,
                                       @SpringBean("wristbandTemplate") WristbandTemplate wristbandTemplate,
                                       @SpringBean("printerService") PrinterService printerService,
                                       @SpringBean("paperRecordService") PaperRecordService paperRecordService,
-                                      UiSessionContext uiSessionContext) throws UnableToPrintLabelException {
+                                       @SpringBean Config config,
+                                       UiSessionContext uiSessionContext) throws UnableToPrintLabelException {
         try {
 
             if (location == null) {
                 location = uiSessionContext.getSessionLocation();
             }
 
-            // make sure a paper record has been created
-            if (!paperRecordService.paperRecordExistsForPatient(patient, location)) {
-                paperRecordService.createPaperRecord(patient, location);
+            String data = "";
+            if (config.getCountry().equals(Country.SIERRA_LEONE)) {
+                SLWristbandTemplate slWristbandTemplate = Context.getRegisteredComponent("slWristbandTemplate", SLWristbandTemplate.class);
+                data = slWristbandTemplate.generateWristband(patient, location, uiSessionContext.getLocale());
+            } else {
```

Comment:
Now that we are printing wristbands for multiple countires, I'd make ZL explicit as well, ie rename WristbandTemplate to ZLWristbandTemplate, get rid of the autowiring of the WristbandTemplate in the code above, and just fetch the "zlWristbandTemplate" if the country = HAITI, the same way we do for Sierra Leone

Reply:
Thanks @mogoodrich ! I have refactored the ZlWristbandTemplate.